### PR TITLE
add LPIPS

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?

Fixes https://github.com/PyTorchLightning/metrics/issues/427
Adds LPIPS metric to image package. Use to measure perceptual similarity.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
   - Permutation Invariant Training (PIT) ([#294](https://github.com/PyTorchLightning/metrics/issues/294))
 
+  - LPIPS ([#431](https://github.com/PyTorchLightning/metrics/issues/431))
```
Comment (Borda) [inline CHANGELOG.md:30]: does it have full name?

File: tests/image/test_lpips.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+)
+
+
+def _compare_fn(img1: Tensor, img2: Tensor, net_type: str, reduction: str = "mean") -> Tensor:
+    ref = reference_LPIPS(net=net_type)
+    res = ref(img1, img2).detach().cpu().numpy()
+    if reduction == "mean":
+        return res.mean()
+    else:
+        return res.sum()
```
Comment (Borda) [inline tests/image/test_lpips.py:43]:
```suggestion
    return res.sum()
```

File: torchmetrics/image/lpip_similarity.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+             before returning the value at the step
+         process_group:
+             Specify the process group on which synchronization is called.
+             default: ``None`` (which selects the entire world)
+         dist_sync_fn:
+             Callback that performs the allgather operation on the metric state. When ``None``, DDP
+             will be used to perform the allgather
+
+    References:
+         [1] The Unreasonable Effectiveness of Deep Features as a Perceptual Metric
```
Comment (Borda) [inline torchmetrics/image/lpip_similarity.py:70]: lest use https://docutils.sourceforge.io/docs/user/rst/quickref.html#citations

File: CHANGELOG.md (hunk 1 above)
Comment (SkafteNicki) [reply CHANGELOG.md:30]: added :]

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
